### PR TITLE
Alertmanager: Update the prometheus/alertmanager to its latest version

### PR DESCRIPTION
**Pull request by gotjosh**
#### What this PR does

Upgrades alertmanager to its latest version.

#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/mimir/issues/2651, which was fixed via https://github.com/prometheus/alertmanager/pull/2968
Fixes https://github.com/grafana/mimir/issues/2806 
A revert of https://github.com/grafana/mimir/pull/2924, which was fixed via https://github.com/prometheus/alertmanager/pull/3080

#### Checklist

- N/A Tests updated
- N/A Documentation added
- N/A `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by gotjosh** [comment]
@pracucci could you please take a look? 🙏 Would love if this could make it to next week's build.

**Comment by pracucci** [comment]
Sure @gotjosh. Looking into it.

**Comment by pracucci** [comment]
```diff
- 	github.com/aws/aws-sdk-go v1.44.66 // indirect
+	github.com/aws/aws-sdk-go v1.44.94 // indirect
```

I went through the [AWS release notes](https://github.com/aws/aws-sdk-go/releases) and everything looks good to me.

File: pkg/alertmanager/alertmanager.go [hunk 1]
```diff
@@ -359,17 +359,21 @@ func (am *Alertmanager) ApplyConfig(userID string, conf *config.Config, rawCfg s
 		return nil
 	}
 
-	muteTimes := make(map[string][]timeinterval.TimeInterval, len(conf.MuteTimeIntervals))
+	timeIntervals := make(map[string][]timeinterval.TimeInterval, len(conf.MuteTimeIntervals)+len(conf.TimeIntervals))
```
Comment (gotjosh) [inline pkg/alertmanager/alertmanager.go:362]: I'm not sure if this warrants a test - I can't seem to find any precedence for such; any ideas?

**Review by pracucci — APPROVED** [review]
LGTM!

File: pkg/alertmanager/alertmanager.go (hunk 1 above)
Comment (pracucci) [reply pkg/alertmanager/alertmanager.go:362]: Ideally yes, but I wouldn't block on it, given the muted times weren't tested even before.

BTW, thanks for fixing this!

Comment (gotjosh) [reply pkg/alertmanager/alertmanager.go:362]: Sounds good, I filed https://github.com/grafana/mimir/issues/3100 to follow up on it.

**Pull request merged**